### PR TITLE
[Gnosis] Initial trade flow

### PR DESCRIPTION
**Pull request by evgenyboxer**
Initial trade flow.

For simplicity, there are lots of files with `GP.vue` to handle the Gnosis version of the file. These may be merged into one file with some conditional logic.

There is also some code that is a straight fork from Cowswap which may be refactored later on.

The whole trade interface is hidden under the env variable: `VUE_APP_GNOSIS_INTEGRATION` (once enabled, it will default to `Gnosis` and will also allow you to go back to `Balancer` via the app nav menu)

Currently only MainNet is supported.

### CODE REVIEW

File: src/constants/env.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const ENV = process.env.VUE_APP_ENV || 'development';
+
+const IS_DEV = ENV === 'development';
+const IS_STAGING = ENV === 'staging';
+const IS_PROD = ENV === 'production';
+
+export { IS_DEV, IS_PROD, IS_STAGING, ENV };
```
Comment (evgenyboxer) [inline src/constants/env.ts:7]: I this its pretty handy to this ^

File: src/lib/config/rinkeby.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "exchangeProxy": "0x4e67bf5bD28Dd4b570FBAFe11D0633eCbA2754Ec",
+    "merkleRedeem": "0x3bc73D276EEE8cA9424Ecb922375A0357c1833B3",
+    "multicall": "0x3b2A02F22fCbc872AF77674ceD303eb269a46ce3",
+    "authorizer": "0xA331D84eC860Bf466b4CdCcFb4aC09a1B43F3aE6",
+    "vault": "0xBA12222222228d8Ba445958a75a0704d566BF2C8",
+    "weightedPoolFactory": "0x8E9aa87E45e92bad84D5F8DD1bff34Fb92637dE9",
+    "stablePoolFactory": "",
+    "tokenFactory": "0xEaa391544B268018C0FA848204E984Bd7FF7f20c",
+    "weth": "0xc778417e063141139fce010982780140aa0cd5ab",
+    "balancerHelpers": "0x94905e703fEAd7f0fD0eEe355D267eE909784e6d"
```
Comment (evgenyboxer) [inline src/lib/config/rinkeby.json:26]: This is just an initial config file, these addresses are not correct.

File: src/services/gnosis/signing.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
+import { APP_NETWORK_ID } from '@/constants/network';
```
Comment (evgenyboxer) [inline src/services/gnosis/signing.ts:1]: This file is mostly a fork from Cowswap

File: src/services/gnosis/operator.service.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        `${this.baseURL}/markets/${market}/${kind}/${amount}`
+      );
+      return response.data;
+    } catch (e) {
+      console.log(`[Gnosis Operator]: Failed to get price from API`, e);
+    }
+
+    return null;
+  }
+}
```
Comment (evgenyboxer) [inline src/services/gnosis/operator.service.ts:108]: This file is mostly a fork from Cowswap

File: src/constants/network.ts [hunk 5]
```diff
@@ -1 +1,11 @@
-export type NetworkId = 1 | 42;
+export type NetworkId = 1 | 3 | 4 | 5 | 42;
+
+export enum Network {
+  MAINNET = 1,
+  ROPSTEN = 3,
+  RINKEBY = 4,
+  GÖRLI = 5,
+  KOVAN = 42
```
Comment (evgenyboxer) [inline src/constants/network.ts:8]: I actually really liked how this was done on Uniswap, so I adopted it too. (Having named network ids)

**Review by garethfuller — APPROVED** [review]
Looks good to merge, doesn't seem to affect the existing app.

File: src/services/gnosis/constants.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    : 'https://protocol-mainnet.gnosis.io/api',
+  [Network.RINKEBY]: IS_DEV
+    ? 'https://protocol-rinkeby.dev.gnosisdev.com/api'
+    : 'https://protocol-rinkeby.gnosis.io/api'
+};
+
+export const OPERATOR_URL =
+  OPERATOR_URLS[APP_NETWORK_ID] ?? OPERATOR_URLS[Network.MAINNET];
+
+export const GP_SETTLEMENT_CONTRACT_ADDRESS = GPv2Settlement[APP_NETWORK_ID]
```
Comment (folex) [inline src/services/gnosis/constants.ts:20]: for me, this fails on Kovan if I run it locally:
```
Uncaught TypeError: Cannot read property 'address' of undefined
    at Module.cd49 (constants.ts:21)
    at c (bootstrap:84)
    at Object.0 (app.b3566afb.js:1)
    at c (bootstrap:84)
    at n (bootstrap:45)
    at bootstrap:221
    at app.b3566afb.js:1
```

not sure if problem is on my side

**Pull request merged**